### PR TITLE
kafka 1.0 has support for time based offsets.

### PR DESCRIPTION
**Pull request by borissmidt**
#75 
This is a pull request that implements a possible solution to the kafka timebased offset.

### CODE REVIEW

File: kafka-1.0.x/src/main/scala/monix/kafka/config/StartFrom.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  case object FromLatest extends StartFrom
+
+  /** Uses the current offset or falls back on the [[AutoOffsetReset]] setting
+    */
+  case object FromCommited extends StartFrom
+
+  /** Resets the offset to an offset starting from now minus the period
+    * @param period the period which we will seek back from
+    */
+  case class PriorToNow(period: FiniteDuration) extends StartFrom
```
Comment (Avasil) [inline kafka-1.0.x/src/main/scala/monix/kafka/config/StartFrom.scala]: could be `final` I think

File: kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala [hunk 2]
```diff
@@ -428,7 +429,7 @@ object KafkaConsumerConfig {
       retryBackoffTime = config.getInt("retry.backoff.ms").millis,
       observableCommitType = ObservableCommitType(config.getString("monix.observable.commit.type")),
       observableCommitOrder = ObservableCommitOrder(config.getString("monix.observable.commit.order")),
-      observableSeekToEndOnStart = config.getBoolean("monix.observable.seekEnd.onStart"),
+      observableStartFrom = if (config.getBoolean("monix.observable.seekEnd.onStart")) FromLatest else FromCommited,
```
Comment (Avasil) [inline kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala]: We could do something similar to `ObservableCommitOrder` and `ObservableCommitType` and create `apply` method.
We could also consider allowing all the options using `conf` file (probably with two parameters)

**Comment by Avasil** [comment]
Thanks @borissmidt - do you think it could be tested?

File: kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala (hunk 2 above)
Comment (borissmidt) [reply kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala]: I actually think these are things that should be configured on a per consumer basis, maybe even on a topic level.

**Comment by borissmidt** [comment]
Yes it could be tested, But most of it is kafka's public api. So i think the test should mainly check the time calculation part and not the actual api call.

File: kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala (hunk 2 above)
Comment (Avasil) [reply kafka-1.0.x/src/main/scala/monix/kafka/KafkaConsumerConfig.scala]: Ok, makes sense. :)

**Comment by Avasil** [comment]
Sorry @borissmidt I think I didn't get a notification for commit
Let me know if you're still interested about this feature

**Comment by borissmidt** [comment]
Oh, no its not a problem, we don't really use this library currently at work and i didn't find the time to make a working unit test during several months so it is better to close the PR then leaving it unfinished.

**Pull request closed without merging**